elephant/signal_processing.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+"""
+docstring goes here.
+
+:copyright: Copyright 2015 by the Elephant team, see AUTHORS.txt.
+:license: Modified BSD, see LICENSE.txt for details.
+"""
+
+import numpy as np
+import scipy.signal
+import quantities as pq
+import neo
+
+
+def butter(anasig, highpassfreq=None, lowpassfreq=None, order=4,
+           filtfunc='filtfilt'):
+    """
+    Butterworth filtering function for AnalogSignal objects. Filter type is
+    determined according to how values of highpassfreq and lowpassfreq are
+    given (see Parameters for details).
+
+    Parameters
+    ----------
+    anasig: AnalogSignal
+        time series data to be filtered
+    highpassfreq, lowpassfreq: Quantity or None
+        high-pass and low-pass cut-off frequencies, respectively.
+        Filter type is determined depending on values of these arguments:
+            highpassfreq only (lowpassfreq = None):    highpass filter
+            lowpassfreq only (highpassfreq = None):    lowpass filter
+            highpassfreq < lowpassfreq:    bandpass filter
+            highpassfreq > lowpassfreq:    bandstop filter
+    order: int
+        Order of Butterworth filter. Default is 4.
+    filtfunc: string
+        Filtering function to be used. Either 'filtfilt'
+        (scipy.signal.filtfilt) or 'lfilter' (scipy.signal.lfilter). In most
+        applications 'filtfilt' should be used, because it doesn't bring about
+        phase shift. Default is 'filtfilt'.
+
+    Returns
+    -------
+    anasig_out: AnalogSignal
+        Filtered signal
+    """
+
+    def apply_filter(signal, Fs, highpassfreq=None, lowpassfreq=None, order=4,
```

Comment:
Is this a public function? If so it needs a docstring (docstring would be good anyway). If it is an internal function, the name should start with an underscore.